src/sentry/integrations/slack/views/unlink_identity.py

```diff
@@ -35,37 +41,69 @@ class SlackUnlinkIdentityView(BaseView):
     """
 
+    _METRICS_SUCCESS_KEY = "sentry.integrations.slack.unlink_identity_view.success"
+    _METRICS_FAILURE_KEY = "sentry.integrations.slack.unlink_identity_view.failure"
+
     @method_decorator(never_cache)
-    def handle(self, request: HttpRequest, signed_params: str) -> HttpResponse:
+    def dispatch(self, request: HttpRequest, signed_params: str) -> HttpResponseBase:
         try:
             params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+        except (SignatureExpired, BadSignature) as e:
+            logger.warning("dispatch.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 request=request,
             )
 
-        organization, integration, idp = get_identity_or_404(
-            ExternalProviders.SLACK,
-            request.user,
-            integration_id=params["integration_id"],
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+            logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+            metrics.incr(self._METRICS_SUCCESS_KEY + ".get_identity", sample_rate=1.0)
+            params.update({"organization": organization, "integration": integration, "idp": idp})
+        except Http404:
+            logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_identity", sample_rate=1.0)
+            raise
+
+        return super().dispatch(request, params=params)
+
+    def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        organization, integration = params["organization"], params["integration"]
+
+        return render_to_response(
+            "sentry/auth-unlink-identity.html",
+            request=request,
+            context={"organization": organization, "provider": integration.get_provider()},
         )
 
-        if request.method != "POST":
-            return render_to_response(
-                "sentry/auth-unlink-identity.html",
-                request=request,
-                context={"organization": organization, "provider": integration.get_provider()},
-            )
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        method = ".post."
+        params = kwargs["params"]
+        integration, idp = params["integration"], params["idp"]
 
         try:
             Identity.objects.filter(idp_id=idp.id, external_id=params["slack_id"]).delete()
         except IntegrityError:
             logger.exception("slack.unlink.integrity-error")
```

Comment:
hopefully this captures/logs the ids we need/want during debugging

Reply:
lol it doesn't, ill add it